subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/NativeBuildPerformanceTest.groovy

```diff
@@ -41,14 +41,17 @@ class NativeBuildPerformanceTest extends AbstractCrossVersionPerformanceTest {
         result.assertCurrentVersionHasNotRegressed()
 
         where:
-        testProject        | maxMemory    | iterations
-        "smallNative"      | '256m'       | 40
-        "mediumNative"     | '256m'       | null
-        "bigNative"        | '1g'         | null
-        "multiNative"      | '256m'       | null
-        "smallCppLibrary"  | '256m'       | 40
-        "mediumCppLibrary" | '256m'       | null
-        "bigCppLibrary"    | '256m'       | null
+        testProject      | maxMemory    | iterations
+        "smallNative"    | '256m'       | 40
+        "mediumNative"   | '256m'       | null
+        "bigNative"      | '1g'         | null
+        "multiNative"    | '256m'       | null
+        "smallCppApp"    | '256m'       | 40
+        "mediumCppApp"   | '256m'       | null
+        "bigCppApp"      | '256m'       | null
+        "smallCppMulti"  | '256m'       | 40
+        "mediumCppMulti" | '256m'       | null
+        "bigCppMulti"    | '1g'         | null
```

Comment:
Does the performance test coordinator run an entire class or does it dispatch each possible test to separate agents?

If the entire class runs remotely (and serially), then we may want to split this into two classes.  It looks like this must be one of the longer sets of tests:
https://builds.gradle.org/viewLog.html?order=DURATION_DESC&recordsPerPage=20&filterText=NativeBuild&status=&currentGroup=test&scope=*%23teamcity%23*%23teamcity%23*&pager.currentPage=1&buildTypeId=Gradle_Check_PerformanceTestCoordinator&buildId=8675238&tab=testsInfo

Reply:
The performance test coordinator dispatches each possible test scenario to a different agent already. These are just some really long-running tests.